### PR TITLE
Special grey door requirements: Morph Ball Room

### PR DESCRIPTION
**Pull request by fdtaillefer**

### CODE REVIEW

File: region/brinstar/blue.json [hunk 1]
```diff
@@ -135,38 +131,84 @@
...
-                {"obstacle":{
-                  "id": "A",
-                  "requires": [
-                    {"or":[
-                      "Bombs",
-                      {"and":[
-                        "PowerBomb",
+                {"or":[
+                  {"cost":[
+                    {"cost":[
```
Comment (DIProgan) [inline region/brinstar/blue.json]: The double cost lines

File: region/brinstar/blue.json [hunk 2]
```diff
@@ -135,38 +131,84 @@
...
+                            "ScrewAttack",
+                            "Plasma"
+                          ]}
+                        ]},
+                        {"and":[
+                          {"cost":[
+                            {"enemyDamage": {
+                              "enemy": "Sidehopper",
+                              "type": "contact",
+                              "hits": 2
```
Comment (DIProgan) [inline region/brinstar/blue.json]: I think it's perfectly fair to expect getting hit only once when entering the room here and then morph down into the corner below while the PB is going off.

File: region/brinstar/blue.json [hunk 3]
```diff
@@ -135,38 +131,84 @@
...
                         {"cost":[
-                          {"ammo": [
-                            {
-                              "type": "PowerBomb",
-                              "count": 1
-                            }
+                          {"enemyDamage": {
+                            "enemy": "Sidehopper",
+                            "type": "contact",
+                            "hits": 8
```
Comment (DIProgan) [inline region/brinstar/blue.json]: This one is tough to balance. I'm unsure about it but 8 does seem a bit excessive. https://youtu.be/7l-mDFOryYc

File: region/brinstar/blue.json (hunk 1 above)
Comment (fdtaillefer) [reply region/brinstar/blue.json]: Dammit

File: region/brinstar/blue.json (hunk 2 above)
Comment (fdtaillefer) [reply region/brinstar/blue.json]: And you can do the ol' morph ball juke with the other guy. Sold.

File: region/brinstar/blue.json (hunk 3 above)
Comment (fdtaillefer) [reply region/brinstar/blue.json]: Yeah it's intended as a "not-quite-worst-case-but-things-didn't-go-well" scenario.

Comment (fdtaillefer) [reply region/brinstar/blue.json]: I don't know to what extent it matters, though. This can literally never be logically required.

Comment (fdtaillefer) [reply region/brinstar/blue.json]: How about 6 instead?

Comment (DIProgan) [reply region/brinstar/blue.json]: Sure

**Comment by DIProgan** [comment]
Everything looks dandy

**Pull request merged**